[struct_pack] Is copy constructor required for deserializing?

### Search before asking

- [ ] I searched the [issues](https://github.com/alibaba/yalantinglibs/issues) and found no similar issues.


### What happened + What you expected to happen

I got a compile error when deserializing a type with copy constructor removed:

> struct_pack.hpp:347:10: error: call to implicitly-deleted copy constructor of  ....


### Reproduction way


### Anything else


### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

--- a/include/ylt/struct_pack/reflection.hpp
+++ b/include/ylt/struct_pack/reflection.hpp
@@ -2208,22 +2208,22 @@ if constexpr (Idx == I) {\
 }\
 
 #define STRUCT_PACK_GET_INDEX(Idx, Type) \
-auto& STRUCT_PACK_GET_##Idx(Type& c) {\
+inline auto& STRUCT_PACK_GET_##Idx(Type& c) {\
     return STRUCT_PACK_GET<STRUCT_PACK_FIELD_COUNT_IMPL<Type>()-1-Idx>(c);\
 }\
 
 #define STRUCT_PACK_GET_INDEX_CONST(Idx, Type) \
-const auto& STRUCT_PACK_GET_##Idx(const Type& c) {\
+inline const auto& STRUCT_PACK_GET_##Idx(const Type& c) {\
     return STRUCT_PACK_GET<STRUCT_PACK_FIELD_COUNT_IMPL<Type>()-1-Idx>(c);\
 }\
 
 #define STRUCT_PACK_REFL(Type,...) \
-Type& STRUCT_PACK_REFL_FLAG(Type& t) {return t;} \
+inline Type& STRUCT_PACK_REFL_FLAG(Type& t) {return t;} \
 template<typename T> \
 constexpr std::size_t STRUCT_PACK_FIELD_COUNT_IMPL(); \
 template<> \
 constexpr std::size_t STRUCT_PACK_FIELD_COUNT_IMPL<Type>() {return STRUCT_PACK_ARG_COUNT(__VA_ARGS__);} \
-decltype(auto) STRUCT_PACK_FIELD_COUNT(const Type &){ \
+inline decltype(auto) STRUCT_PACK_FIELD_COUNT(const Type &){ \
   return std::integral_constant<std::size_t,STRUCT_PACK_ARG_COUNT(__VA_ARGS__)>{}; \
 } \
 template<std::size_t I> auto& STRUCT_PACK_GET(Type& c) { \

--- a/src/struct_pack/tests/test_struct.hpp
+++ b/src/struct_pack/tests/test_struct.hpp
@@ -112,3 +112,9 @@ struct my_tree {
              *a.right_child == *b.right_child));
   }
 };
+
+struct person_with_refl {
+  std::string name;
+  int age;
+};
+STRUCT_PACK_REFL(person_with_refl, name, age);

